File: deliveryAppUI/src/app/models/vehicle.ts
export class Vehicle {
  vehicleId: number;
  vehicleModelId: number;
  plate: string;
  yearManufacture: number;
  createAt: string;
  model: VehicleModel;

  constructor(
    vehicleId: number,
    vehicleModelId: number,
    plate: string,
    yearManufacture: number,
    createAt: string,
    model: VehicleModel
  ) {
    this.vehicleId = vehicleId;
    this.vehicleModelId = vehicleModelId;
    this.plate = plate;
    this.yearManufacture = yearManufacture;
    this.createAt = createAt;
    this.model = model;
  }
}

export class VehicleModel {
  modelId: number;
  model: string;
  brand: Brand;
  name: string;

  constructor(modelId: number, model: string, brand: Brand, name: string) {
    this.modelId = modelId;
    this.model = model;
    this.name = name;
    this.brand = brand;
  }
}

export class Brand {
  brandId: number;
  brand: string;
  name: string;

  constructor(brandId: number, brand: string, name: string) {
    this.brandId = brandId;
    this.name = brand;
    this.brand = brand;
  }
}

File: DeliveryApp/src/DeliveryApp.Domain/Entity/Vehicle.cs
using DeliveryApp.Domain.Validation;
using Flunt.Notifications;

namespace DeliveryApp.Domain.Entity
{
	public class Vehicle : Notifiable<Notification>
	{
		public int VehicleId { get; set; }
		public int VehicleModelId { get; set; }
		public string Plate { get; set; }
		public int YearManufacture { get; set; }
		public DateTime CreateAt { get; set; }

		public VehicleModel VehicleModel { get; set; }
		public bool IsRent { get; set; }

		public Vehicle()
		{

		}

		public Vehicle(int vehicleId, int vehicleModelId, string plate, int yearManufacture)
		{
			VehicleId = vehicleId;
			VehicleModelId = vehicleModelId;
			Plate = plate.ToUpper();
			YearManufacture = yearManufacture;
			CreateAt = DateTime.SpecifyKind(DateTime.UtcNow, DateTimeKind.Utc);

			AddNotifications(new VehicleValidator(this));
		}
	}
}
